pkg/grpc/server.go

```diff
@@ -179,16 +181,21 @@ func (a *apiImpl) Stop() bool {
 }
 
 func (a *apiImpl) unaryInterceptors() []grpc.UnaryServerInterceptor {
+	var u []grpc.UnaryServerInterceptor
+
 	// The metrics and error interceptors are first in line, i.e., outermost, to
 	// make sure all requests are registered in Prometheus with errors converted
 	// to gRPC status codes.
-	u := []grpc.UnaryServerInterceptor{
-		grpc_prometheus.UnaryServerInterceptor,
-		grpc_errors.ErrorToGrpcCodeInterceptor,
-		contextutil.UnaryServerInterceptor(a.requestInfoHandler.UpdateContextForGRPC),
-		contextutil.UnaryServerInterceptor(authn.ContextUpdater(a.config.IdentityExtractors...)),
+	u = append(u, grpc_prometheus.UnaryServerInterceptor)
+	u = append(u, grpc_errors.ErrorToGrpcCodeInterceptor)
+
+	if a.config.RateLimiter != nil {
```

Comment:
I think in the current state of the code, `a.config.RateLimiter` is never nil (at worst it has a rate if `Inf`).

Reply:
It can be `nil` - an example is `scanner/cmd/scanner/main.go` <- when the Scanner creates a server. Currently, we are adding RateLimiter only for central.